src/Xml/Dom/Configurator/canonicalize.php

```diff
@@ -5,29 +5,23 @@
 namespace VeeWee\Xml\Dom\Configurator;
 
 use Closure;
-use DOMDocument;
+use \DOM\XMLDocument as DOMDocument;
 use VeeWee\Xml\Dom\Document;
 use function Psl\Type\non_empty_string;
 use function VeeWee\Xml\Dom\Loader\xml_string_loader;
-use function VeeWee\Xml\ErrorHandling\disallow_libxml_false_returns;
 
 /**
  * @return Closure(DOMDocument): DOMDocument
  */
 function canonicalize(): Closure
 {
     return static fn (DOMDocument $document): DOMDocument
-        => Document::configure(
-            pretty_print(),
-            loader(
-                xml_string_loader(
-                    disallow_libxml_false_returns(
-                        non_empty_string()->assert($document->C14N()),
-                        'Could not canonicalize XML input.'
-                    ),
-                    LIBXML_NSCLEAN + LIBXML_NOCDATA
-                )
+        => Document::fromLoader(
+            xml_string_loader(
+                non_empty_string()->assert($document->C14N()),
+                LIBXML_NSCLEAN + LIBXML_NOCDATA
             ),
+            pretty_print(),
```

Comment:
TODO : Not sure it's a pretty print or canonicalize issue, but the format has shifted to 3 spaces with multiple spaces. It might be solved once pretty-print works again.

Reply:
Actually sorry, this is not a bug but related to the preserveWhiteSpace thing.
See https://github.com/veewee/xml/pull/74#discussion_r1501190089 it _is_ possible but I forgot this constant was already exposed. Changing the code so the constant is passed to the parser fixes this test case.

Reply:
That works! Thanks